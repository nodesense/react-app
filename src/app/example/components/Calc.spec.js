/**
 * @fileOverview: Test suite for the Calculator component.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Calc from './Calc';

describe('Calc component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <Calc />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the correct sum', () => {
    const calculator = new Calc();

    const sum = calculator._getSum(3, 5);
    expect(sum).toEqual(8);
  });

  it('should render required form elements', () => {
    const calculator = shallow(
      <Calc />
    );

    const form = calculator.find('form');
    expect(form.length).toBe(1);
    expect(form.find('input').length).toBe(2);
    expect(form.find('button').length).toBe(1);
    expect(form.find('p.result').length).toBe(1);
  });

  it('should display the result on add', () => {
    const calculator = mount(
      <Calc />
    );

    const form = calculator.find('form');

    form.childAt(0).instance().value = 3;
    form.childAt(1).instance().value = 5;
    form.find('button').simulate('click');

    const result = calculator.find('.result');
    expect(result.text()).toEqual('8');
  });
});