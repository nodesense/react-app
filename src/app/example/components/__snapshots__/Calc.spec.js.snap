// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calc component should render snapshot 1`] = `
<form>
  <input
    defaultValue={0}
    type="number"
  />
  <input
    defaultValue={0}
    type="number"
  />
  <button
    onClick={[Function]}
    type="button"
  />
  <p
    className="result"
  />
</form>
`;
